// @2chaos/webpack_react_todolist/dist/src/App.js
var __spreadArray = (this && this.__spreadArray) || function (to, from, pack) {
    if (pack || arguments.length === 2) for (var i = 0, l = from.length, ar; i < l; i++) {
        if (ar || !(i in from)) {
            if (!ar) ar = Array.prototype.slice.call(from, 0, i);
            ar[i] = from[i];
        }
    }
    return to.concat(ar || Array.prototype.slice.call(from));
};
import "./index.css";
import React, { useState } from "react";
import { createRoot } from "react-dom/client";
import { Button, Space } from "antd";
import { Input } from "antd";
import { List } from "antd";
function App() {
    function change() {
        setTodo(__spreadArray(__spreadArray([], todo, true), [state], false));
        updateState("");
    }
    function onDelete(index) {
        todo.splice(index, 1);
        setTodo(__spreadArray([], todo, true));
    }
    function swapItem(index1, index2) {
        var temp = todo[index1];
        todo[index1] = todo[index2];
        todo[index2] = temp;
        setTodo(__spreadArray([], todo, true));
    }
    var _a = useState([]), todo = _a[0], setTodo = _a[1];
    var _b = useState(""), state = _b[0], updateState = _b[1];
    var _c = useState(), clickPageNum = _c[0], setClickPageNum = _c[1];
    var total = 0;
    var _d = useState(function () {
        var todoList = [];
        for (var index = 0; index < 100; index++) {
            total++;
            todoList.push({
                title: "title" + index,
                desc: "desc" + index,
            });
        }
        return todoList;
    }), todo1 = _d[0], setTode = _d[1];
    /*
     */
    var pagesize = 5;
    var _e = useState(function () {
        var i = (total + pagesize - 1) / pagesize;
        while (1) {
            if (i % 1 === 0) {
                return i;
            }
            else {
                total = total - 1;
                i = (total + pagesize - 1) / pagesize;
            }
        }
    }), pageNum = _e[0], setPageNum = _e[1];
    var _f = useState("bottom"), position = _f[0], setPosition = _f[1];
    var _g = useState("end"), align = _g[0], setAlign = _g[1];
    //Math.floor((total + pagesize - 1) / pagesize));
    return (React.createElement("article", null,
        React.createElement("h1", null, "Todo"),
        React.createElement(Space, null,
            React.createElement(Input, { style: { width: 1293 }, placeholder: "\u8BF7\u8F93\u5165", value: state, onChange: function (e) {
                    updateState(e.target.value);
                }, onKeyDown: function (e) {
                    if (e.key === "Enter") {
                        setTodo(__spreadArray(__spreadArray([], todo, true), [state], false));
                        updateState("");
                    }
                } }),
            React.createElement(Space, { wrap: true },
                React.createElement(Button, { type: "dashed", onClick: function () {
                        change();
                    } }, "Orz"))),
        React.createElement(List, { style: { marginTop: 24 }, header: React.createElement("div", null, "Header"), footer: React.createElement("div", null, "Footer"), bordered: true, dataSource: todo, renderItem: function (item) { return React.createElement(List.Item, null, item); }, pagination: {
                position: position,
                align: align,
                defaultCurrent: 1,
                total: todo.length,
                pageSize: 5,
            } })));
}
var container = document.getElementById("2Chaos");
var root = createRoot(container);
root.render(React.createElement(App, null));
export default App;
